s2a/src/main/java/io/grpc/s2a/internal/handshaker/SslContextFactory.java

```diff
@@ -63,7 +63,8 @@ static SslContext createForClient(
           KeyStoreException,
           NoSuchAlgorithmException,
           UnrecoverableKeyException,
-          GeneralSecurityException {
+          GeneralSecurityException,
+          IllegalArgumentException {
```

Comment:
This isn't right. There was not an invalid argument passed in to this method. This exception needs to be handled within the function and converted into a more meaningful error. At the very least it should be clear that the error was because we received something unknown from the s2a server.

Often for this sort of thing, converting it to some form of IOException is good.

Reply:
Done in [0b1e0bf](https://github.com/grpc/grpc-java/pull/11588/commits/0b1e0bf790f29aaa3b2e0ffd61d26945fa2d46db)